sql/src/main/java/io/druid/sql/calcite/rel/DruidSemiJoin.java

```diff
@@ -235,6 +236,22 @@ protected RelDataType deriveRowType()
   }
 
   @Override
+  public void replaceInput(int ordinalInParent, RelNode p)
+  {
+    if (ordinalInParent == 0) {
+      Preconditions.checkArgument(
+          p instanceof DruidRel,
+          "RelNode[%s] is not a DruidRel", p.getClass().getCanonicalName()
+      );
+      this.left = (DruidRel<?>) p;
```

Comment:
Left is not really an input (it sort of "becomes" this rel). See getInputs which only returns 1 element as far as Calcite is concerned. So this should only replace `right`.

Reply:
Thanks. Fixed.